vulnerabilities/pipelines/samba_importer.py

```diff
@@ -0,0 +1,335 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# VulnerableCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/aboutcode-org/vulnerablecode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+import html
+import logging
+import re
+from datetime import datetime
+from traceback import format_exc as traceback_format_exc
+from typing import Dict
+from typing import Iterable
+from typing import List
+from typing import Optional
+
+import pytz
+import requests
+from bs4 import BeautifulSoup
+from dateutil import parser as dateparser
+from packageurl import PackageURL
+
+from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importer import AffectedPackage
+from vulnerabilities.importer import Reference
+from vulnerabilities.pipelines import VulnerableCodeBaseImporterPipeline
+from vulnerabilities.utils import build_description
+from vulnerabilities.utils import get_cwe_id
+
+
+class SambaImporterPipeline(VulnerableCodeBaseImporterPipeline):
+    pipeline_id = "samba_importer"
+    spdx_license_expression = "GPL-3.0-only"
+    license_url = "https://www.samba.org/samba/ms_license.html"
+    url = "https://www.samba.org/samba/history/security.html"
+    importer_name = "Samba Importer"
+
+    @classmethod
+    def steps(cls):
+        return (cls.fetch, cls.collect_and_store_advisories, cls.import_new_advisories)
+
+    def fetch(self):
+        self.log(f"Fetching `{self.url}`")
+        try:
+            response = requests.get(self.url)
+            response.raise_for_status()
+            self.advisory_data = response.text
+            self.advisory_details_cache = {}
+        except requests.exceptions.HTTPError as http_err:
+            self.log(
+                f"Failed to fetch Samba security data: {self.url} with error {http_err!r}:\n{traceback_format_exc()}",
+                level=logging.ERROR,
+            )
+            raise
+
+    def advisories_count(self):
+        if not hasattr(self, "advisory_data"):
+            return 0
```

Comment:
This will not be required, if `fetch` step fails for some reason entire pipeline would be terminated.